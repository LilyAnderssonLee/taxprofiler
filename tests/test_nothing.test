nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should run without failures") {

        expect {
            assert workflow.success

            assert file("${outputDir}/multiqc/multiqc_report.html").readLines().join(' ').matches(".* for a more heavy duty testing suite.*")
            assert file("${outputDir}/fastqc/raw/2612_ERR5766176_B_raw_1_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2612_ERR5766176_B_raw_2_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2613_ERR5766181_raw_2_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2612_ERR5766176_raw_1_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2612_ERR5766176_raw_2_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/ERR3201952_ERR3201952_raw_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2613_ERR5766181_raw_1_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/fastqc/raw/2612_ERR5766180_raw_fastqc.html").readLines().join(' ').matches(".*version.*")
            assert file("${outputDir}/pipeline_info/software_versions.yml").readLines().join(' ').matches(".*version.*")

            //assert snapshot(path("${outputDir}/pipeline_info/database_sheet.valid.csv")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc.log")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_overrepresented_sequences_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Percentages.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_sequence_length_distribution_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc_general_stats.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_per_sequence_quality_scores_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc_fastqc.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_adapter_content_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_per_base_n_content_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_sequence_counts_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Counts.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_sequence_duplication_levels_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc_data.json")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/mqc_fastqc_per_base_sequence_quality_plot_1.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc_citations.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_data/multiqc_sources.txt")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Percentages.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Counts.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_duplication_levels_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_adapter_content_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_overrepresented_sequences_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_length_distribution_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_n_content_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1_pc.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_quality_scores_plot_1.svg")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_sequence_length_distribution_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Counts.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_adapter_content_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1_pc.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_overrepresented_sequences_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_sequence_duplication_levels_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Percentages.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/png/mqc_fastqc_per_base_n_content_plot_1.png")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_sequence_quality_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_length_distribution_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_adapter_content_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_overrepresented_sequences_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_duplication_levels_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Counts.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_n_content_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_quality_scores_plot_1.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1_pc.pdf")).match()
            //assert snapshot(path("${outputDir}/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Percentages.pdf")).match()
        }

    }

}
