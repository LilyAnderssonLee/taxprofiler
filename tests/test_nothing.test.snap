{
    "Should run without failures": {
        "content": [
            "2612_ERR5766176_B_raw_1_fastqc.html:md5,da77f755624291c33dedae9ae1a34009"
        ],
        "timestamp": "2022-10-27T11:36:31+0000"
    }
}